packages/client/components/DashTopBar.tsx

```diff
@@ -89,7 +89,20 @@ const TopBarMain = styled('div')({
 })
 
 const DashTopBar = (props: Props) => {
-  const {toggle, viewer} = props
+  const {toggle, viewerRef} = props
+  const data = useFragment(
+    graphql`
+      fragment DashTopBar_viewer on Query {
```

Comment:
Thanks for pointing it out, I also added a TODO here to the commit https://github.com/ParabolInc/parabol/pull/6530/files#diff-ac6c3d215db84985036cd25e4f95b42d879189780bb09c0277bad90a568fbbf1R24-R25

currently can't `on User` directly because User type does not implement Node (i.e. a type that has an id). ref: https://relay.dev/docs/api-reference/use-refetchable-fragment/#arguments

actually I tried to add the `Node GraphQLInterfaceType` but it seems not to work, so for now I leave it here and limit the scope of this refactoring.

```ts
import {GraphQLID, GraphQLInterfaceType, GraphQLNonNull} from 'graphql'

const NodeInterface: GraphQLInterfaceType = new GraphQLInterfaceType({
  name: 'Node',
  description: 'An object with an ID',
  fields: () => ({
    id: {
      type: new GraphQLNonNull(GraphQLID),
      description: 'The id of the object.'
    }
  })
})

export default NodeInterface
```

```TypeScript
const User: GraphQLObjectType<any, GQLContext> = new GraphQLObjectType<any, GQLContext>({
  name: 'User',
  description: 'The user account profile',
  interfaces: () => [NodeInterface],
})
```

Reply:
@JimmyLv what exactly doesn't work about this for the `useFragment` case?  I would expect existing fragments that are ` on User` in `createFragmentContainer` to also work with `useFragment`.

`usePaginationFragment` does need the `refetchable` directive, but `useFragment` should be fine.

Reply:
@JimmyLv what exactly doesn't work about this for the `useFragment` case?  I would expect existing fragments that are ` on User` in `createFragmentContainer` to also work with `useFragment`.

`usePaginationFragment` does need the `refetchable` directive, but `useFragment` should be fine.

Reply:
@JimmyLv I haven't forgotten about this PR, I'm trying to figure out what could be wrong. 

My latest findings are that our `Notification` type isn't formally implementing the `Node` type and seems like Relay directly asks for that. Here's a similar problem: https://stackoverflow.com/questions/71616913/react-relay-invalid-use-of-refetchable-on-fragment 

The issue with our `Notification` type is that it is an interface type itself, and even though there's an option provide interfaces like this:
```
const Notification: GraphQLInterfaceType = new GraphQLInterfaceType({
  name: 'Notification',
  fields: () => notificationInterfaceFields,
  interfaces: () => [NodeInterface],
```
it's not being added in generated schema ([can interface implements interface in graphql?](https://stackoverflow.com/questions/57888591/interface-extends-multiple-interfaces-in-graphql-schema)), where Notification looks like
```
interface Notification {
  """A shortid for the notification"""
  id: ID!

```

I tried adding `Node` interface to each of the notification types, it's being added correctly in schema
```
"""
A notification alerting the user that they have been promoted (to team or org leader)
"""
type NotifyPromoteToOrgLeader implements Notification & Node {
```
but seems like it's not enough for Relay to stop complaining. 

I'll continue to investigate this, sorry it's taking so long!

Reply:
I've encountered this as well, I wrote about it under paginated queries here: https://www.parabol.co/blog/upgrade-react-to-graphql-relay-hooks/

Here's a video on how to debug something like this: https://www.loom.com/share/05157500e42940b49a02b5bbd038aafb

What do you think is the best way to solve this? List out some pros/cons & we'll go from there

Reply:
At this point, I'm not sure. I suspect the issue is the usage of graphql interface in the place where relay expects an object that implements `Node` interface. In graphql interface can not implement other interfaces so this would probably mean redesigning the schema somehow which isn't worth the effort. A similar problem was described here: https://stackoverflow.com/questions/71616913/react-relay-invalid-use-of-refetchable-on-fragment

Reply:
i'm OK with passing through things that fragment on type Query, but if we do that, we should rename the fragments. i'll call out an example of where it needs to be fixed